src/model/TickDataSource.ts

```diff
@@ -57,10 +60,32 @@ export default class TickDataSource extends MongoDataSource<TickType> {
    * @returns the new/updated tick
    */
   async editTick (filter: TickEditFilterType, updatedTick: TickInput): Promise<TickType | null> {
+    await this.validateTick(updatedTick)
     const rs = await this.tickModel.findOneAndUpdate(filter, updatedTick, { new: true })
     return await rs?.toObject() ?? null
   }
 
+  private async validateTick (tick: TickInput): Promise<void> {
+    const climbIdAsUUID = muuid.from(tick.climbId)
+    const climb = await this.climbModel.findOne({ _id: climbIdAsUUID, _deleting: { $eq: null } }).lean()
+    if (climb == null) {
+      throw new Error('Climb not found')
+    }
+    // Some imports from MP have both a route and boulder grade for example The heart route on el cap
+    const isBoulderingOnly = (climb.type.bouldering === true) && Object.keys(climb.type).every(key => key === 'bouldering' || climb.type[key] === false)
+    // bouldering only. solo, lead, follow, tr only applicable to roped climbs
+    if (isBoulderingOnly) {
+      if ((!['Send', 'Flash', 'Attempt'].includes(tick.attemptType)) || (['Lead', 'Solo', 'Tr', 'Follow'].includes(tick.style))) {
+        throw new Error('Invalid attempt type or style for bouldering')
+      }
+    // Only applicable to boulders or roped climbs with "lead" style
+    } else if (!(tick.style === 'Lead')) {
+      if ((['Attempt', 'Redpoint', 'Pinkpoint', 'Flash', 'Onsight'].includes(tick.attemptType))) {
+        throw new Error('Invalid attempt type for a non-lead style')
+      }
+    }
+  }
```

Comment:
This really gets in the weeds pretty fast, but since we have the option of `Fell/Hung` for Lead climbs, isn't that redundant with `Attempt`? 

On MP, `Attempt` can only be marked for boulders, and `Fell/Hung` can only be marked for roped climbs with 'Lead' style.

Reply:
Actually I think we shouldn't support Fell/Hung. Keep in mind, these are not necessarily the user-facing values. I think attempt should map to Fell/Hung for climbs on the user-facing side, but stored as attempt in the server.